File: NCAA/scripts/_check.mjs
import fs from 'fs';

// Check NCAA model for Purdue
const h = JSON.parse(fs.readFileSync('data/history.json','utf8'));
const runs = h.runs || [];
const today = runs.filter(r => r.date === '20260315' || r.dateDisplay === '2026-03-15');
if (!today.length) {
  console.log('No NCAA run for today');
} else {
  for (const r of today) {
    for (const g of (r.games || [])) {
      const name = (g.home + ' ' + g.away).toLowerCase();
      if (name.includes('purdue')) {
        console.log('=== PURDUE (NCAA Model) ===');
        console.log(`  ${g.away} @ ${g.home}`);
        console.log(`  Line: ${g.line}  Total: ${g.total}`);
        console.log(`  Proj: Away ${g.aS}  Home ${g.hS}  pT: ${g.pT}`);
        console.log(`  Margin: ${g.margin}  sDiff: ${g.sDiff}`);
        console.log(`  P(home cover): ${g.pHomeCover}  P(away cover): ${g.pAwayCover}`);
        console.log(`  sPick: ${g.sPick}  sConf: ${g.sConf}`);
        console.log(`  oPick: ${g.oPick}`);
        console.log(`  dNET: ${g._features?.dNET?.toFixed(1)}`);
      }
    }
    // Also show all picks for today
    console.log('\n=== All NCAA picks today ===');
    for (const g of (r.games || [])) {
      if (g.sPick !== 'PASS') {
        console.log(`  ${g.sPick} (P=${g.pCover}) — ${g.away} @ ${g.home}`);
      }
    }
  }
}

File: PythonNCAA/scripts/_check.py
#!/usr/bin/env python3
# scripts/_check.py — Check NCAA model for specific team
import json

h = json.loads(open("data/history.json").read())
runs = h.get("runs") or []
today = [r for r in runs if r.get("date") == "20260315" or r.get("dateDisplay") == "2026-03-15"]
if not today:
    print("No NCAA run for today")
else:
    for r in today:
        for g in (r.get("games") or []):
            name = (g.get("home", "") + " " + g.get("away", "")).lower()
            if "purdue" in name:
                print("=== PURDUE (NCAA Model) ===")
                print(f"  {g['away']} @ {g['home']}")
                print(f"  Line: {g.get('line')}  Total: {g.get('total')}")
                print(f"  Proj: Away {g.get('aS')}  Home {g.get('hS')}  pT: {g.get('pT')}")
                print(f"  Margin: {g.get('margin')}  sDiff: {g.get('sDiff')}")
                print(f"  P(home cover): {g.get('pHomeCover')}  P(away cover): {g.get('pAwayCover')}")
                print(f"  sPick: {g.get('sPick')}  sConf: {g.get('sConf')}")
                print(f"  oPick: {g.get('oPick')}")
                features = g.get("_features") or {}
                print(f"  dNET: {features.get('dNET', 0):.1f}" if features.get("dNET") else "  dNET: N/A")
        print("\n=== All NCAA picks today ===")
        for g in (r.get("games") or []):
            if g.get("sPick") != "PASS":
                print(f"  {g.get('sPick')} (P={g.get('pCover')}) -- {g.get('away')} @ {g.get('home')}")
